src/sys/unix/uds/listener.rs

```diff
@@ -103,11 +95,24 @@ pub(crate) fn accept(listener: &net::UnixListener) -> io::Result<(UnixStream, So
         Ok(s)
     });
 
-    socket
-        .map(UnixStream::from_std)
-        .map(|stream| (stream, SocketAddr::from_parts(sockaddr, socklen)))
-}
+    let socket = socket.map(UnixStream::from_std)?;
 
-pub(crate) fn local_addr(listener: &net::UnixListener) -> io::Result<SocketAddr> {
-    super::local_addr(listener.as_raw_fd())
+    #[allow(unused_mut)] // See below.
+    let mut path_len = socklen as usize - path_offset(&sockaddr);
+    // Darwin is being weird, it return a length of 16, but other an unnamed
+    // (all zero) address. Map that to a length of 0 to match other OS.
```

Comment:
```suggestion
    // Darwin is being weird, it returns a length of 16, but otherwise an unnamed
    // (all zero) address. Map that to a length of 0 to match other OS.
```